skill-tree /src/main/java/com/RDS/skilltree/utils/JWTUtils.java

```diff
@@ -0,0 +1,79 @@
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+@Component
+public class JWTUtils {
+
+    @Value("${jwt.rds.public.key}")
+    private String publicRDSKeyString;
+
+    /**
+     * Converts the given private key string to an RSAPrivateKey object.
+     *
+     * @param privateKeyString The private key string to be converted.
+     * @return The corresponding RSAPrivateKey object.
+     * @throws Exception If an error occurs during the conversion process.
+     */
+    private RSAPrivateKey convertToRSAPrivateKey(String privateKeyString) throws Exception {
+        try {
+            privateKeyString = privateKeyString.replace("-----BEGIN PRIVATE KEY-----", "")
+                    .replace("-----END PRIVATE KEY-----", "")
+                    .replaceAll("\\s", "");
+
+            byte[] privateKeyBytes = Base64.getDecoder().decode(privateKeyString);
+            PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(privateKeyBytes);
+            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            return (RSAPrivateKey) keyFactory.generatePrivate(keySpec);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException | IllegalArgumentException e) {
+            throw new Exception("Error converting private key: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Converts the given public key string to an RSAPublicKey object.
+     *
+     * @param publicKeyString the public key string to be converted
+     * @return the RSAPublicKey object converted from the public key string
+     * @throws Exception if there is an error during the conversion process
+     */
+    private RSAPublicKey convertToRSAPublicKey(String publicKeyString) throws Exception {
+        publicKeyString = publicKeyString.replace("-----BEGIN PUBLIC KEY-----", "")
+                .replace("-----END PUBLIC KEY-----", "")
+                .replaceAll("\\s", "");
+
+        byte[] publicKeyBytes = Base64.getDecoder().decode(publicKeyString);
+        X509EncodedKeySpec keySpec = new X509EncodedKeySpec(publicKeyBytes);
+        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+        return (RSAPublicKey) keyFactory.generatePublic(keySpec);
+    }
+
+    public String getRDSUserId(String token) throws Exception {
+        Claims claims = Jwts.parser().setSigningKey(convertToRSAPublicKey(publicRDSKeyString)).parseClaimsJws(token).getBody();
+        String temp = claims.get("userId", String.class);
+        return temp;
+    }
+
+    public boolean validateToken(String token) throws Exception {
```

Comment:
Use a catch block to handle specific exceptions. The Jwts.parser().parseClaimsJws() method throws a JwtException if the token is invalid or expired. To handle this exception, you can use a catch block. This will allow you to return a more specific error message to the caller.

Use a return statement instead of a throw statement. If the token is valid, the Jwts.parser().parseClaimsJws() method will return a Claims object. You can use a return statement to return true to the caller if the token is valid and false otherwise.

Reply:
Yes Tejas we are catching the error in the caller method and the error with custom message is sent to the user. The message will be `Invalid JWT`